src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -190,10 +194,10 @@ private async Task ProcessQueryAsync(ReindexJobQueryStatus query, CancellationTo
 
                 // TODO: Release lock on job document so another thread may pick up the next query.
 
-                await _updateIndices.ProcessSearchResultsAsync(results, _reindexJobRecord.Hash, cancellationToken);
+                var extractRequest = new ExtractReindexRequest(results, _reindexJobRecord.Hash);
+                await _mediator.Send(extractRequest, cancellationToken);
```

Comment:
Curiosity, why use the mediator instead of making an interface for the reindex utilities and calling process search results here?

Reply:
That is a good suggestion.  I was simply separating the ReindexTask from the code that calls the Extractor.  You are correct an interface would be a more straightforward method of separation.